protocols/v2/codec-sv2/src/decoder.rs

```diff
@@ -203,7 +200,7 @@ impl<T: Serialize + binary_sv2::GetSize, B: IsBuffer> WithoutNoise<B, T> {
             0 => {
                 self.missing_b = Header::SIZE;
                 let src = self.buffer.get_data_owned();
-                let frame = Sv2Frame::<T, B::Slice>::from_bytes_unchecked(src);
+                let frame = Sv2Frame::<T, B::Slice>::from_bytes(src)?;
```

Comment:
we can use unchecked here, if you are concerned you can put a check in debug mode